runtime/autoload/man.vim

```diff
@@ -181,36 +202,28 @@ endfunction
 " attempt to extract the name and sect out of 'name(sect)'
 " otherwise just return the largest string of valid characters in ref
 function! s:extract_sect_and_name_ref(ref) abort
-  if a:ref[0] ==# '-' " try ':Man -pandoc' with this disabled.
-    throw 'manpage name cannot start with ''-'''
+  if empty(a:ref)
+    throw 'Empty argument.'
   endif
-  let ref = matchstr(a:ref, '[^()]\+([^()]\+)')
-  if empty(ref)
-    let name = matchstr(a:ref, '[^()]\+')
-    if empty(name)
-      throw 'manpage reference cannot contain only parentheses'
-    endif
-    return ['', name]
-  endif
-  let left = split(ref, '(')
+
+  let ml = matchlist(a:ref, '^\(.\{-1,}\)\((\(.\{-1,}\))\|\.\([^.]\{-1,}\)\)\?$')
```

Comment:
Could you justify this as well? This regex is a beast and I can't tell what it's improving.

Reply:
man deals with Unix names: pretty much any (ASCII?) character is valid in a Unix (file)name. So I updated the matching to accept, e.g., "`--a_bad_idea(9BadIdeas)`" as `name = '--a_bad_idea'` and `sect = '9BadIdeas'`. That's why I needed to add `--` to the arguments of the `s:system()` / `s:run_man()` function.

The regex matches man refs of the form `foo(bar)` and `foo.bar`, and if neither matches then it matches the whole string (which was checked to be non-empty). Here `foo` and `bar` are valid Unix characters for (file)names (except for NULL and new lines).